lib/widgets/weekday.dart

```diff
@@ -0,0 +1,109 @@
+import 'package:concordia_navigation/models/course.dart';
+import 'package:concordia_navigation/providers/map_data.dart';
+import 'package:concordia_navigation/services/size_config.dart';
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:concordia_navigation/storage/app_constants.dart' as constants;
+import 'package:provider/provider.dart';
+
+class Weekday extends StatelessWidget {
+  String weekday;
+  List<Course> courseList;
+
+  Weekday(String weekday, List<Course> courseList) {
+    this.weekday = weekday;
+    this.courseList = courseList;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    DateFormat formatter = DateFormat("Hm");
+
+    List<Widget> courseContainer = [];
+    courseContainer.add(Text(weekday, style: TextStyle(fontSize: 20.0)));
+
+    courseList.forEach((course) => courseContainer.add(ListTile(
+        contentPadding: EdgeInsets.only(
+            left: SizeConfig.safeBlockHorizontal * 5.0,
+            right: SizeConfig.safeBlockHorizontal * 5.0),
+        leading: CircleAvatar(backgroundImage: AssetImage('assets/logo.png')),
+        title: Text(
+          course.summary,
+        ),
+        subtitle: Row(mainAxisSize: MainAxisSize.min, children: <Widget>[
+          Text(
+            "${formatter.format(course?.start?.toLocal())} - ${formatter.format(course?.end?.toLocal())} | ",
+          ),
+          Text((course == null || course?.location == '')
+              ? "N/A"
```

Comment:
Issue in `ios` when course location is NULL.
Add the following the resolve `course?.location == null`

Reply:
is there an all purpose empty check in dart that we can use?

Reply:
IOS calendar keeps it `null` so we'll have to check that option as well.

Reply:
https://stackoverflow.com/questions/42446566/dart-null-false-empty-checking-how-to-write-this-shorter

Reply:
Done